### PR TITLE
[risk=low][no ticket] Fix the access-renewal UI guard when RT renewable modules are incomplete

### PR DESCRIPTION
**Pull request by jmthibault79**
Description:

---
**PR checklist**

- [ ] This PR meets the Acceptance Criteria in the JIRA story
- [ ] The JIRA story has been moved to Dev Review
- [ ] This PR includes appropriate unit tests
- [ ] I have added explanatory comments where the logic is not obvious
- [ ] I have run and tested this change locally, and my testing process is described here
- [ ] If this includes a new feature flag, I have created and linked new JIRA tickets to (a) turn on the feature flag and (b) remove it later
- [ ] If this includes an API change, I have run the E2E tests on this change against my local server with [yarn test-local](https://github.com/all-of-us/workbench/blob/master/e2e/README.md#examples) because this PR won't be covered by the CircleCI tests 
- [ ] If this includes a UI change, I have taken screen recordings or screenshots of the new behavior and notified the PO and UX designer in [Slack](https://pmi-engteam.slack.com/archives/C02MWP2RN5P)
- [ ] If this change impacts deployment safety (e.g. removing/altering APIs which are in use) I have documented these in the description
- [ ] If this includes an API change, I have updated the appropriate Swagger definitions and updated the appropriate API consumers
  * AoU UI
  * [Perf tests](https://github.com/broadinstitute/mcnulty/blob/develop/src/test/scala/services/AoU.scala)
  * [Researcher Directory export](https://github.com/all-of-us/workbench/wiki/Researcher-Directory-(RDR-export))
  * Cron tasks - for Offline*Controllers
  * SumoLogic - for EgressAlert

### CODE REVIEW

File: ui/src/app/pages/access/access-renewal.tsx [hunk 1]
```diff
@@ -154,20 +156,6 @@ export const RenewalRequirementsText = () => (
   </span>
 );
 
-// Helper Functions
-
-// is the module "renewal complete" ?
-// meaning (bypassed || (complete and not expiring))
-const isRenewalCompleteForModule = (status: AccessModuleStatus) => {
```
Comment (jmthibault79) [inline ui/src/app/pages/access/access-renewal.tsx:161]: moved to access-utils

File: ui/src/app/routing/guards.tsx [hunk 2]
```diff
@@ -46,8 +47,7 @@ export const registrationGuard: Guard = {
 };
 
 export const expiredGuard: Guard = {
-  allowed: (): boolean =>
-    !profileStore.get().profile.accessModules.anyModuleHasExpired,
```
Comment (jmthibault79) [inline ui/src/app/routing/guards.tsx:50]: `anyModuleHasExpired` is an old concept from the early days of AAR, and has not kept up with our logic.  We should eventually remove it from the API.

File: ui/src/app/routing/guards.tsx [hunk 3]
```diff
@@ -46,8 +47,7 @@ export const registrationGuard: Guard = {
 };
 
 export const expiredGuard: Guard = {
```
Comment (jmthibault79) [inline ui/src/app/routing/guards.tsx:49]: TODO: rename to rtExpiredGuard?

File: ui/src/app/utils/access-utils.spec.tsx [hunk 4]
```diff
@@ -438,13 +438,11 @@ describe('hasExpired', () => {
   });
 
   it('should return hasExpired=false for a null date', () => {
-    const testTime = nowPlusDays(null);
```
Comment (jmthibault79) [inline ui/src/app/utils/access-utils.spec.tsx:441]: noticed that these were weird - updated

File: ui/src/app/utils/access-utils.tsx [hunk 5]
```diff
@@ -300,8 +300,8 @@ export const getAccessModuleConfig = (
   );
 };
 
-// the modules subject to Annual Access Renewal (AAR), in the order shown on the AAR page.
-export const accessRenewalModules = [
+// the modules subject to Registered Tier Annual Access Renewal (AAR), in the order shown on the AAR page.
+export const rtAccessRenewalModules = [
```
Comment (jmthibault79) [inline ui/src/app/utils/access-utils.tsx:304]: lots of opportunities for refactoring here, like deriving this list from module config.  But today I'm going for expediency, to get this into the next release.

File: ui/src/app/utils/access-utils.tsx [hunk 6]
```diff
@@ -486,6 +486,31 @@ export const isExpiringOrExpired = (
...
+  const wasBypassed = !!status?.bypassEpochMillis;
+  return (
+    wasBypassed ||
+    (isComplete &&
+      !isExpiringOrExpired(status?.expirationEpochMillis, status.moduleName))
+  );
+};
+
+export const isRtRenewalComplete = (profile: Profile): boolean => {
+  const modules = profile?.accessModules?.modules;
```
Comment (jmthibault79) [inline ui/src/app/utils/access-utils.tsx:502]: is there a better alternative?  I could do something like `const {modules} = profile?.accessModules ?? {}` but that doesn't really seem to be an improvement

**Review by aweng98 — APPROVED** [review]
Reviewed in mob meeting this morning. No concerns from me.

**Pull request merged**